src/app/data-planes/views/DataPlaneDetailView.vue

```diff
@@ -36,36 +36,44 @@
       </template>
 
       <DataSource
-        v-slot="{ data, isLoading, error }: DataplaneOverviewSource"
-        :src="`/meshes/${route.params.mesh}/dataplane-overviews/${route.params.dataPlane}`"
+        v-slot="{ data, error }: DataplaneSource"
+        :src="`/meshes/${route.params.mesh}/dataplanes/${route.params.dataPlane}`"
       >
-        <LoadingBlock v-if="isLoading" />
+        <DataSource
+          v-slot="{ data: dataplaneOverview, error: dataplaneOverviewError }: DataplaneOverviewSource"
+          :src="`/meshes/${route.params.mesh}/dataplane-overviews/${route.params.dataPlane}`"
+        >
+          <ErrorBlock
+            v-if="error"
+            :error="error"
+          />
 
-        <ErrorBlock
-          v-else-if="error"
-          :error="error"
-        />
+          <ErrorBlock
+            v-else-if="dataplaneOverviewError"
+            :error="dataplaneOverviewError"
+          />
 
-        <EmptyBlock v-else-if="data === undefined" />
+          <LoadingBlock v-else-if="data === undefined || dataplaneOverview === undefined" />
 
-        <DataPlaneDetails
-          v-else
-          :dataplane-overview="data"
-          data-testid="detail-view-details"
-        />
+          <DataPlaneDetails
+            v-else
+            :dataplane="data"
+            :dataplane-overview="dataplaneOverview"
+            data-testid="detail-view-details"
+          />
```

Comment:
So in practice we only need the Dataplane from the source when we show the resource block (so when the overview is shown). To me it seems that the Datasource should be there where we actually use the data not around everything like here

Reply:
Good point, I’ve moved this to wrap around the code block.